packages/std/src/init_handle.rs

```diff
@@ -10,12 +11,15 @@ use crate::types::HumanAddr;
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
-pub enum CosmosMsg {
+// See https://github.com/serde-rs/serde/issues/1296 why we cannot add trait bounds to T
+pub enum CosmosMsg<T = NativeMsg>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
     Bank(BankMsg),
-    // this is dangerous to use, as it ties you to one particular runtime format.
-    // this makes the contract non-portable, and also fragile to break upon a hardfork
-    // only safe way is to receive it from a user and hold it temporarily.
-    Native(NativeMsg),
+    // by default we use NativeMsg, but a contract can override that
+    // to call into more app-specific code (whatever they define)
+    Native(T),
```

Comment:
Using raw messages as a fallback for no registered custom messages feels a bit unsymmetric. 

What about moving the raw version into `CosmosMsg::Unknown(Binary)`, where Unknown refers to "unknown module".

Then `CosmosMsg<T = NativeMsg>` can fall back to some kind of `NoMsg` type (like TypeScript's `never`) and custom modules go into `CosmosMsg::Custom(T)`?

Reply:
I like this. Made some changes in that direction, but yes, using `CosmosMsg::Native(NoMsg)` as default is good.

Do I even want to add the "unknown"? Blockchains can choose to support a `Raw` variant in their `CustomMsg`.

Reply:
> Do I even want to add the "unknown"? Blockchains can choose to support a `Raw` variant in their `CustomMsg`.

Right, even better. This is added explicitely by developers who know what they are doing and does not need to be supported on all chains.

Reply:
I think I addressed the `NoMsg` part here: https://github.com/CosmWasm/cosmwasm/pull/295/commits/4563b166af7c3c1b45583bf7d7a570d658cec6f7 

Do you want me to add the `CosmosMsg::Unknown` variant? I would rather not unless there is a clear purpose. But I am open to arguments.

Reply:
Perfect. With the example in `reflect/src/msg.rs` it gets clear that we don't need to add it to cosmwasm-std